### PR TITLE
Add a page which lists all the deals

### PR DESCRIPTION
**Pull request by limli**
Screenshot 2020-07-20 at 11 53 32 AM

### CODE REVIEW

File: src/test/java/com/google/step/TestConstants.java [hunk 1]
```diff
@@ -127,4 +127,28 @@
...
+
+  public static final String DEAL_A_BRIEF_JSON =
+      String.format(
+          "{"
+              + "    \"id\": %d,"
+              + "    \"description\": \"%s\","
+              + "    \"image\": \"%s\","
+              + "    \"restaurant\": %d,"
+              + "    \"votes\": %d,"
+              + "    \"poster\": %d"
```
Comment (jyx11011) [inline src/test/java/com/google/step/TestConstants.java]: Are we removing votes, restaurants and poster from deal brief json?

**Review by jyx11011 — COMMENTED** [review]
Good idea to add a page to list all deals! maybe we can add similar pages for other entities as well

File: src/test/java/com/google/step/TestConstants.java (hunk 1 above)
Comment (limli) [reply src/test/java/com/google/step/TestConstants.java]: oyea i think we are. Which PR is it in?

**Review by jyx11011 — DISMISSED** [review]

File: src/main/java/com/google/step/servlets/DealPostListServlet.java [hunk 2]
```diff
@@ -23,14 +23,14 @@
 
 /** Servlet that handles posting deals. */
 @WebServlet("/api/deals")
-public class DealPostServlet extends HttpServlet {
+public class DealPostListServlet extends HttpServlet {
```
Comment (SuyashLakhotia) [inline src/main/java/com/google/step/servlets/DealPostListServlet.java]: Just `DealServlet`?

File: src/test/java/com/google/step/servlets/DealPostListServletTest.java [hunk 3]
```diff
@@ -26,47 +29,38 @@
...
 import org.powermock.api.mockito.PowerMockito;
 import org.powermock.core.classloader.annotations.PrepareForTest;
 import org.powermock.modules.junit4.PowerMockRunner;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
 
 @RunWith(PowerMockRunner.class)
 @PrepareForTest(ImageUploader.class)
-public class DealPostServletTest {
+public class DealPostListServletTest {
```
Comment (SuyashLakhotia) [inline src/test/java/com/google/step/servlets/DealPostListServletTest.java]: Ditto

**Review by SuyashLakhotia — CHANGES_REQUESTED** [review]

**Review by SuyashLakhotia — APPROVED** [review]

**Pull request merged**